lvsfunc/util.py

```diff
@@ -231,31 +185,28 @@ def replace_ranges(clip_a: vs.VideoNode,
     :param use_plugin:      Use the ReplaceFramesSimple plugin for the rfs call (Default: True).
 
     :return:                Clip with ranges from clip_a replaced with clip_b.
+
+    :raises ValueError:     A string is passed instead of a list of ranges.
     """
     if ranges is None:
         return clip_a
 
     if isinstance(ranges, str):  # type:ignore[unreachable]
         raise ValueError("replace_ranges: 'This function does not take strings! Please use a list of tuples or ints!")
 
+    nranges = normalize_ranges(clip_b, ranges)
+
+    if use_plugin and hasattr(core, 'remap'):
```

Comment:
There is a discrepancy in the function between use_plugin=True and False.
Namely, the mismatch; better way to handle it, if you want no mismatch is to set it to false and move the check at L205 before this call, if you want mismatch, replace the trim an splice calls that are done with python notation with std calls and mismatch=True